### PR TITLE
Add recipe how to update blocks programmatically

### PR DESCRIPTION
**Pull request by texnixe**

### CODE REVIEW

File: content/1_docs/2_cookbook/3_templating/0_update-blocks-programmatically/recipe.txt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    'text'  => 'This is a heading',
+    'level' => 'h2',
+  ],
+  'type'     => 'heading',
+]);
+
+$quote = new Kirby\Cms\Block( [
+  'content' => [
+    'text'     => 'Time flies like an arrow; fruit flies like a banana',
+    'citation' => 'a href=\"https://en.wikipedia.org/wiki/Anthony_Oettinger\" rel=\"noopener noreferrer nofollow\">Anthony Oettinger</a>',
```
Comment (afbora) [inline content/1_docs/2_cookbook/3_templating/0_update-blocks-programmatically/recipe.txt]: Missing start `<a` tag.

Comment (texnixe) [reply content/1_docs/2_cookbook/3_templating/0_update-blocks-programmatically/recipe.txt]: Thanks!

**Review by afbora — APPROVED** [review]

**Review by lukasbestle — COMMENTED** [review]
To be honest I think this is another example of a missing core feature. Manually encoding the data as JSON doesn't feel right...

**Comment by afbora** [comment]
Yes, I absolutely agree, the `->toJson()` method like `->toArray()` can be added to the block class.

**Comment by lukasbestle** [comment]
I don't think that's the right way. I was thinking more about a method to update a `blocks` field directly. Same for `structure` actually. This way everything will still work if we change the way these fields store their data.

**Comment by texnixe** [comment]
Great idea, structures have always been a pain to update all these years. Problem is, as always, that people need a solution now, not in x months, years ...

**Comment by afbora** [comment]
Ahh, wait a minute. This way already supports both array and objects 😅 

````php
// array
$page->update([
  'text' => $blocks->toArray()
]);

// object
$page->update([
  'text' => $blocks
]);
````

https://github.com/getkirby/kirby/blob/3.5.0/src/Form/Field/BlocksField.php#L179-L183

**Review by lukasbestle — APPROVED** [review]

**Pull request merged**